lib/plugins/command_block.js

```diff
@@ -1,14 +1,26 @@
 const assert = require('assert')
-
 const ProtoDef = require('protodef').ProtoDef
 
 module.exports = inject
 
 function inject (bot) {
-  function setCommandBlock (pos, command, trackOutput, mode) {
+  function setCommandBlock (pos, command, options = {}) {
+    assert.strictEqual(bot.player.gamemode, 1, new Error('The bot has to be in creative mode to open the command block window'))
     assert.notStrictEqual(pos, null)
     assert.notStrictEqual(command, null)
-    trackOutput = trackOutput || 2
+    assert.strictEqual(bot.blockAt(pos).name, 'command_block', new Error("The block isn't a command block"))
+
+    // Default values when a command block is placed in vanilla minecraft
+    options.trackOutput = options.trackOutput || false
+    options.conditional = options.conditional || false
+    options.alwaysActive = options.alwaysActive || false
+    options.mode = options.mode || 2 // Possible values: 0: SEQUENCE, 1: AUTO and 2: REDSTONE
+
+    let flags = 0
+    if (!options.trackOutput) flags += 1
+    if (options.conditional) flags += 2
+    if (options.alwaysActive) flags += 4
```

Comment:
Comments on this would be appreciated

Reply:
Add flags:

```js
const FLAG_TRACK_OUTPUT = 1 << 0
const FLAG_CONDITIONAL = 1 << 1
const FLAG_ALWAYS_ACTIVE = 1 << 2
```
Just a suggestion how you could use them:
```js
// ...
const { trackOutput = false, conditional = false, alwaysActive = false } = options
const flags = (+trackOutput * FLAG_TRACK_OUTPUT) 
  | (+conditional * FLAG_CONDITIONAL) 
  | (+alwaysActive * FLAG_ALWAYS_ACTIVE)

```
or you can:
```js
let flags = 0
if (trackOutput) flags |= FLAG_TRACK_OUTPUT
```

Reply:
Thanks! That's what I was thinking of but didn't know how to do